### PR TITLE
Feature/mixr

### PR DESCRIPTION
**Pull request by nicholas-sly**
Fun with dealing with the dynamic number of cards for different isotopes.  Also, some of the formatting was ambiguous in the instructions and there are no examples in the test problems, but there is an example further down in the MIXR chapter, which provided some clarity.

Changed the approach to the multiple cards from what was partially implemented to something else.  Either way, seems to work.

### CODE REVIEW

File: src/njoy21/input/MIXR/Card3/src/readMatnWtnList.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  std::vector< std::pair< Argument< Matn >, Argument< Wtn > > > matnWtnList;
+  std::vector< int > matnList;
+
+  if( nnin > 10 ){
+    Log::error( "The maximum number of materials allowed is 10." );
+    throw std::exception();
+  }
+
+  for( size_t i = 0; i < nnin; i++ ){
+    auto matn_tmp = argument::extract< MIXR::Card3::Matn >( is );
```
Comment (jlconlin) [inline src/njoy21/input/MIXR/Card3/src/readMatnWtnList.hpp]: `mtn_tmp` and `wtn_tmp` should be defined outside the loop and then just reassigned inside the loop. In this way we avoid the cost of repeatedly constructing those values.

File: src/njoy21/input/MIXR/Card3/test/Card3.test.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#include "njoy21.hpp"
+
+using namespace njoy::njoy21::input;
+
+SCENARIO( "MIXR, Card3",
+          "[MIXR] [Card3]" ){
+  GIVEN( "valid inputs" ){
+    WHEN( "valid values are provided" ){
+      std::string inp(
+ " 816 0.2 /\n"
```
Comment (jlconlin) [inline src/njoy21/input/MIXR/Card3/test/Card3.test.cpp]: I don't think each line of Card3 ends with a `/`. I think only the card (can be multiple lines) is terminated with a `/`. Have you tested this in NJOY2016?

Comment (nicholas-sly) [reply src/njoy21/input/MIXR/Card3/test/Card3.test.cpp]: Page 625 of the NJOY2016 manual has an example case that does have the forward slashes after every pair of (matn,wtn).

Comment (jlconlin) [reply src/njoy21/input/MIXR/Card3/test/Card3.test.cpp]: I saw this after I posted that message.

**Pull request merged**